wordpress-mobile/WordPress-iOS
# Adopt to the breaking changes in WordPressOrgRestApi

## Description

`WordPressOrgRestApi` used to only support self-hosted sites. It's been rewritten to support self hosted sites and WP.com sites. This PR adopts this breaking changes.

The existing `Blog.wordPressOrgRestApi` returns a non-nil reference only when the blog is a self-hosted site, which is not obvious from the property name "wordPressOrgRestApi", because every WordPress site is bundled with [.org rest API](https://developer.wordpress.org/rest-api/). I have renamed it to `selfHostedSiteRestApi`, especially considering `WordPressOrgRestApi` now supports all sites.

## Test Instructions

There are only a number of places in the app that calls .org REST API. Editor settings API is one of them, which will be fired upon entering Pages List and opening a post. Plugin is another place that uses .org REST API, which is only accessible using self hosted site in the app.

You can put a breakpoint in `BlockEditorSettingsService.fetchTheme` function to see the API result. Or, you can use a proxy to see all 'wp/v2' HTTP requests.

## Regression Notes
1. Potential unintended areas of impact

All features that call .org REST API. i.e. plugins and gutenberg settings.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Verified plugins and gutenberg settings API calls work.

3. What automated tests I added (or what prevented me from doing so)

None.

PR submission checklist:


Testing checklist: N/A
